File: 力扣习题/338比特位计数/problems.ts
function countBits(n: number): number[] {
    const res = new Array<number>(n + 1);
    res[0] = 0;
    for(let i = 1; i < n + 1; i++) {
      if(i % 2 == 0) {
        res[i] = res[i / 2];
      } else {
        res[i] = res[i - 1] + 1;
      }
    }
    return res;
};

File: 力扣习题/338比特位计数/problems.py
from typing import List

class Solution:
    def countBits(self, n: int) -> List[int]:
      res = [0] * (n + 1)
      for i in range(n + 1):
        if i % 2 == 0:
          res[i] = res[i // 2]
        else:
          res[i] = res[i - 1] + 1
      return res
